src/altinn-app-frontend/src/features/form/containers/Form.tsx

```diff
@@ -161,14 +149,44 @@ export function Form() {
         spacing={3}
         alignItems='flex-start'
       >
-        {currentView === currentLayout &&
-          filteredLayout &&
-          filteredLayout.map((component) => {
-            return renderLayoutComponent(component, layout);
-          })}
+        {mainComponents.map((component) =>
+          renderLayoutComponent(component, layout),
+        )}
+        <ErrorReport components={errorReportComponents} />
       </Grid>
-    </div>
+    </>
   );
 }
 
-export default Form;
+function topLevelComponents(layout: ILayout) {
+  const inGroup = new Set<string>();
+  layout.forEach((component) => {
+    if (component.type === 'Group') {
+      const childList = component.edit?.multiPage
+        ? component.children.map((childId) => childId.split(':')[1] || childId)
+        : component.children;
+      childList.forEach((childId) => inGroup.add(childId));
+    }
+  });
+  return layout.filter((component) => !inGroup.has(component.id));
+}
+
+function extractBottomButtons(layout: ILayout) {
+  const extract = new Set<ComponentTypes>([
+    'NavigationButtons',
+    'Button',
+    'PrintButton',
+  ]);
+
+  const toMainLayout: ILayout = [];
+  const toErrorReport: ILayout = [];
+  for (const component of [...layout].reverse()) {
+    if (extract.has(component.type) && toMainLayout.length === 0) {
+      toErrorReport.push(component);
+    } else {
+      toMainLayout.push(component);
+    }
+  }
+
+  return [toMainLayout.reverse(), toErrorReport.reverse()];
```

Comment:
Not really sure why these are reversed?

Reply:
Because we only want to extract the buttons etc from the bottom/end of the layout. In the for-loop above the layout is copied and reversed when we loop it so that we can easily extract the bottom-most components, but skip a `Button` located near the top of the layout. Uhh, not sure if that was more understandable, but a unit-test would surely help here - I'll get on it!